### PR TITLE
Add new page to report known issues.

### PR DESCRIPTION
**Pull request by Rombur**

### CODE REVIEW

File: docs/source/known-issues.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Known issues
+
+## HIP
+- Compatibility issue between HIP and gcc 8. You may encounter the following error:
```
[the remark this answers is not in the document]
Comment (Rombur) [reply docs/source/known-issues.rst:4]: The problem is from gcc not from HIP. I don't see how changing the version of HIP will help with that bug.

[the remark this answers is not in the document]
Comment (Rombur) [reply docs/source/known-issues.rst:4]: I disagree. We know that several versions of ROCm do not work with gcc 8, the problem is on the gcc side, there is no reason to believe that the code will somehow work with a newer version of ROCm. If we say explicit which versions of ROCm there is a problem with, we will need to update this file for every new release of ROCm.

[the remark this answers is not in the document]
Comment (dalg24) [reply docs/source/known-issues.rst:4]: Sure

**Review by dalg24 — APPROVED** [review]
Did you put much thinking on whether and where we should have cross references to that page?  For instance from the "requirements" page?

**Comment by Rombur** [comment]
> Did you put much thinking on whether and where we should have cross references to that page? For instance from the "requirements" page?

No, I didn't think about it. I don't have a strong opinion either way.

**Pull request merged**